apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/RetryingMutateRowsTest.cs

```diff
@@ -183,6 +186,88 @@ public class RetryingMutateRowsTest
             Assert.Equal((int)Code.DeadlineExceeded, entry.Status.Code);
         }
 
-        // TODO: Test with a mock IClock to make sure we don't go over the total timeout when retrying
+        [Fact]
+        public async Task RetryAfterTotalExpiration()
+        {
+            var settings = new BigtableServiceApiSettings();
+            // Don't allow for any time to retry.
+            settings.MutateRowsRetrySettings =
+                settings.MutateRowsRetrySettings.WithTotalExpiration(
+                    Expiration.FromTimeout(TimeSpan.Zero));
+
+            var request = new MutateRowsRequest
+            {
+                Entries =
+                {
+                    Mutations.CreateEntry("a", Mutations.DeleteFromRow()),
+                    Mutations.CreateEntry("b", Mutations.DeleteFromRow()),
+                    Mutations.CreateEntry("c", Mutations.DeleteFromRow())
+                }
+            };
+            var client = Utilities.CreateMutateRowsMockClient(
+                request,
+                entriesForInitialStream: new[]
+                {
+                    Utilities.CreateMutateRowsResponseEntry(0, Code.Ok),
+                    Utilities.CreateMutateRowsResponseEntry(1, Code.DeadlineExceeded),
+                    Utilities.CreateMutateRowsResponseEntry(2, Code.Ok)
+                },
+                entriesForRetryStreams: new[]
+                {
+                    // 1st retry response entries
+                    new[] { Utilities.CreateMutateRowsResponseEntry(0, Code.Ok) }
+                },
+                settings: settings);
+
+            bool exceptionOccurred = false;
+            try
```

Comment:
You could use

```csharp
var exception = await Assert.ThrowsAsync<RpcException>(() => client.MutateRowsAsync(request));
Assert.Equal(StatusCode.DeadlineExceeded, ex.StatusCode);
```

I *think* that would be a bit simpler to read, but I don't have much of a preference.

Reply:
Thanks, I didn't realize you could get back the thrown exception from Assert.ThrowsAsync. This is much cleaner.